nixos/modules/services/networking/cloudflare-warp.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.cloudflare-warp;
+in
+{
+  options.services.cloudflare-warp = {
+    enable = mkEnableOption "Cloudflare Zero Trust client daemon";
+
+    package = mkPackageOption pkgs "cloudflare-warp" { };
+
+    rootDir = mkOption {
+      type = types.str;
+      default = "/var/cloudflare-warp";
+      description = ''
+        Working directory for the warp-svc daemon.
+      '';
+    };
+
+    udpPort = mkOption {
+      type = types.port;
+      default = 2408;
+      description = ''
+        The UDP port to open in the firewall. Warp uses port 2408 by default, but fallback ports can be used
+        if that conflicts with another service. See the [firewall documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall#warp-udp-ports)
+        for the pre-configured available fallback ports.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Open the `udpPort` port in the firewall.
+      '';
+    };
```

Comment:
Probably against best-practice to open ports on the firewall by default. Another option might be to leave this default-false, and then add an eval-time warning against none of the fallback ports being allowed in the firewall config.

From the [docs](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall#warp-udp-ports):

> WARP utilizes UDP for all of its communications. By default, the UDP port required for WARP is UDP 2408. WARP can fallback to UDP 500, UDP 1701, or UDP 4500.

Reply:
Sometimes you need to keep things enabled by default so it works out of the box. For example [KDE connect also does it](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/programs/kdeconnect.nix).